src/main/kotlin/fittibackendapp/domain/diet/facade/DietRecordQueryFacade.kt

```diff
@@ -0,0 +1,32 @@
+package fittibackendapp.domain.diet.facade
+
+import fittibackendapp.domain.diet.service.DietDietRecordService
+import fittibackendapp.domain.diet.service.DietRecordService
+import fittibackendapp.domain.diet.service.DietService
+import fittibackendapp.dto.PcfAmountInGramsDto
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class DietRecordQueryFacade (
+    private val dietService: DietService,
+    private val dietRecordService: DietRecordService,
+    private val dietDietRecordService: DietDietRecordService
+){
+    fun getPcfAmountInGramsBetweenDays(userId: Long, fromDate: LocalDate, toDate: LocalDate) : PcfAmountInGramsDto {
+        val dietRecords = dietRecordService.findDietRecordsBetweenTwoDays(userId, fromDate, toDate)
+        var protein = 0.0
+        var carbohydrate = 0.0
+        var fat = 0.0
+        dietRecords.forEach { dietRecord ->
+            val dietDietRecords = dietDietRecordService.findAllByDietRecordId(dietRecord.id)
+            dietDietRecords.forEach {
+                val diet = dietService.findById(it.diet.id)
+                protein += diet.protein * it.weight
+                carbohydrate += diet.carbohydrate * it.weight
+                fat += diet.fat * it.weight
+            }
+        }
```

Comment:
```suggestion
val dietRecordIds = dietRecordService
            .findDietRecordsBetweenTwoDays(userId, fromDate, toDate)
            .map { it.id }

        val dietDietRecords = dietDietRecordService
            .findAllByDietRecordIds(dietRecordIds)
            .forEach {
                val diet = dietService.findById(it.diet.id)
                protein += diet.protein * it.weight
                carbohydrate += diet.carbohydrate * it.weight
                fat += diet.fat * it.weight
            }
````
이렇게하면 쿼리를 덜 보낼 수 있을듯. 지금 있는 forEach도 없어지면 좋을 거같음.

Reply:
그리고 n+1문제 발생할거같음. 근데 그건 나중에 해결해도될듯